openqa-get-job-schedule

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# 
+# Extract a SCHEDULE parameter for the given job
+# 
+
+import json
+import sys
+from openqa_client.client import OpenQA_Client
+
+def get_schedule(instance, job) :
+	'''
+	Build a SCHEDULE parameter from the given job (id) on the given instance
+	'''
+	schedule = []
+	
+	## Configure instance
+	client = OpenQA_Client(server=instance)
+	path = "jobs/%d/details" % (job)
+	job = client.openqa_request(method="GET", path=path)['job']
+	# iterate over testresults. category is only set for a new category (e.g. "installation", "console", ecc.)
+	category = ""
+	for result in job['testresults'] :
+		if "category" in result : category = result['category']
+		name = result['name']
+		# Ignore fragment
+		if '#' in name :
+			i = name.find('#')
+			name = name[:i]
+		schedule.append("%s/%s" % (category, name))
+	return schedule
+
+def is_int(x) :
+	try :
+		x = int(x)
```

Comment:
The assigned value will never the read, right? So I'd rather avoid it.

Of course it would be nice if that assignment would actually be effective. Then the function could be called `try_convert_to_int` and the converted value could be used by the caller (avoiding the 2nd conversion). Maybe something like this can be achieved by using a reference (in Perl it would be very easy, not sure about Python)?

Reply:
In Python you pass objects and integer objects are immutable, so something like a reference to a int does not exists naturally.

Given the rather simple approach to the problem, I also feel that this would be very much overshoot the problem.